projects/core/steps/artifacts.py

```
@@ -0,0 +1,250 @@
+            success=True,  # Never fail - this is a finally step
+            message=message,
+            data={"deleted": deleted, "errors": errors},
+        )
+
+    def _delete_resource(self, kind: str, name: str, ns_args: list[str]) -> bool:
+        """Delete a Kubernetes resource. Returns True if successful."""
+        try:
+            cmd = ["oc", "delete", kind, name, "--ignore-not-found", *ns_args]
+            logger.debug(f"Running: {' '.join(cmd)}")
+
+            result = subprocess.run(
```

Ruff (0.15.9)

[error] `subprocess` call: check for execution of untrusted input

(S603)